owncloudApp/src/androidTest/java/com/owncloud/android/settings/security/OCSettingsPasscode.kt

```diff
@@ -0,0 +1,226 @@
+        onView(withText(R.string.pass_code_mismatch)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun cancelFirstTry() {
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        onView(withId(R.id.txt0)).perform(replaceText("1"))
+        onView(withId(R.id.txt1)).perform(replaceText("1"))
+        onView(withId(R.id.txt2)).perform(replaceText("1"))
+
+        onView(withId(R.id.cancel)).perform(click())
+        assertTrue(errorMessage, activityRule.activity.isFinishing)
+    }
+
+    @Test
+    fun cancelSecondTry() {
+        //Open Activity in passcode creation mode
+        intent.setAction(PassCodeActivity.ACTION_REQUEST_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //First typing
+        typePasscode(arrayOf('1','1','1','1'))
+
+        onView(withId(R.id.txt0)).perform(replaceText("1"))
+        onView(withId(R.id.txt1)).perform(replaceText("1"))
+
+        onView(withId(R.id.cancel)).perform(click())
+        assertTrue(errorMessage, activityRule.activity.isFinishing)
+    }
+
+    @Test
+    fun deletePasscodeView() {
+        //Save a passcode in Preferences
+        storePasscode()
+
+        //Open Activity in passcode deletion mode
+        intent.setAction(PassCodeActivity.ACTION_CHECK_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        onView(withText(R.string.pass_code_remove_your_pass_code)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun deletePasscodeCorrect() {
+        //Save a passcode in Preferences
+        storePasscode("1111")
+
+        //Open Activity in passcode deletion mode
+        intent.setAction(PassCodeActivity.ACTION_CHECK_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //Type correct passcode
+        typePasscode(arrayOf('1','1','1','1'))
+
+        assertTrue(errorMessage, activityRule.activity.isFinishing)
+    }
+
+    @Test
+    fun deletePasscodeIncorrect() {
+        //Save a passcode in Preferences
+        storePasscode("1111")
+
+        //Open Activity in passcode deletion mode
+        intent.setAction(PassCodeActivity.ACTION_CHECK_WITH_RESULT)
+        activityRule.launchActivity(intent)
+
+        //Type incorrect passcode
+        typePasscode(arrayOf('1','1','1','2'))
+
+        onView(withText(R.string.pass_code_enter_pass_code)).check(matches(isDisplayed()))
+    }
+
+    fun typePasscode (digits: Array<Char>){
+        onView(withId(R.id.txt0)).perform(replaceText(digits[0].toString()))
+        onView(withId(R.id.txt1)).perform(replaceText(digits[1].toString()))
+        onView(withId(R.id.txt2)).perform(replaceText(digits[2].toString()))
+        onView(withId(R.id.txt3)).perform(replaceText(digits[3].toString()))
+    }
+
+    fun storePasscode (passcode: String = "1111"){
+        var appPrefs = PreferenceManager.getDefaultSharedPreferences(context).edit();
```

Comment:
This variable is not modified, so it could be `val`

Reply:
Done 👍

Reply:
Can you check this again? It seems to be `var` yet

Reply:
sorry, i did not change it (i pasted the "Done" by error). The point is that the `appPrefs` variable is changed some lines below to add the new preference value

`appPrefs.putBoolean(PassCodeActivity.PREFERENCE_SET_PASSCODE, true)`

Reply:
You are not modifying the variable itself but calling some of those methods that do not touch the appPrefs variable. 

Besides, Android Studio suggest it

![Screenshot 2019-07-01 at 14 36 10](https://user-images.githubusercontent.com/15261363/60437001-a0ee7200-9c0d-11e9-967b-8292546a420e.png)

Reply:
Done 👍